source/jormungandr/tests/chaos_disruptions_tests.py

```diff
@@ -640,17 +640,130 @@ def test_disruption_on_journey(self):
+        assert line_id == 'A'
+
+        # Step 2 : Send an impact on a line with start date same as PT arrival datetime
+        # Proposed application period = (20120614T080102, 20120614T090000)
+        self.send_mock(
+            "blocking_line_disruption_starting_from_pt_arrival",
+            "A",
+            "line",
+            blocking=True,
+            start_period="20120614T080102",
+            end_period="20120614T090000",
+        )
+        # Check disruptions
+        response = self.query_region('disruptions')
+        disruptions = get_not_null(response, 'disruptions')
+        assert len(disruptions) - nb_pre_loaded_disruption == 1
+        for d in disruptions:
+            is_valid_disruption(d)
+        assert "blocking_line_disruption_starting_from_pt_arrival" in [d['disruption_uri'] for d in disruptions]
+        nb_pre_loaded_disruption = len(disruptions)
+
+        # Check that the journey with PT exists and line 'A' is used in PT
+        response = self.query_region(journey_basic_query + "&data_freshness=realtime")
+        assert contains_pt_journey(response) is True
+        pt_sections = get_pt_sections(response)
+        assert len(pt_sections) == 1
+        assert pt_sections[0]['departure_date_time'] == '20120614T080100'
+        assert pt_sections[0]['arrival_date_time'] == u'20120614T080102'
+        line_id = next(o['id'] for o in pt_sections[0]['links'] if o['type'] == 'line')
+        assert line_id == 'A'
+
+        # Step 3 : Send an impact on a line with start date 1 second before PT arrival datetime
+        # Proposed application period = (20120614T080101, 20120614T090000)
+        self.send_mock(
+            "blocking_line_disruption_starting_1s_before_pt_arrival",
+            "A",
+            "line",
+            blocking=True,
+            start_period="20120614T080101",
+            end_period="20120614T090000",
+        )
+        # Check disruptions
+        response = self.query_region('disruptions')
+        disruptions = get_not_null(response, 'disruptions')
+        assert len(disruptions) - nb_pre_loaded_disruption == 1
+        for d in disruptions:
+            is_valid_disruption(d)
+        assert "blocking_line_disruption_starting_1s_before_pt_arrival" in [
+            d['disruption_uri'] for d in disruptions
+        ]
+        nb_pre_loaded_disruption = len(disruptions)
+
+        # Check that the journey with PT exists but line 'M' is used in PT as 'A' is impacted
+        response = self.query_region(journey_basic_query + "&data_freshness=realtime")
+        assert contains_pt_journey(response) is True
+        pt_sections = get_pt_sections(response)
+        assert len(pt_sections) == 1
+        assert pt_sections[0]['departure_date_time'] == '20120614T080101'
+        assert pt_sections[0]['arrival_date_time'] == u'20120614T080103'
+        line_id = next(o['id'] for o in pt_sections[0]['links'] if o['type'] == 'line')
+        assert line_id == 'M'
+
```

Comment:
theses 3 steps test 3 limit : 
- disruption end = journey start
- disruption start = journey end
- disrution start before journey end

one limit isn't tested : 
- disruption end after journey start

Reply:
Will be added